### DISCUSSION TITLE
Saving Order Input to MongoDB

### DISCUSSION THREAD

**Discussion by joshishaswat** [aannirajpatel/samosabucket#7] [Q&A]
We've implemented a radio button that takes a binary input of either "Saturday" or "Sunday" as seen below:

<img width="396" alt="Screen Shot 2021-05-18 at 1 45 24 PM" src="https://user-images.githubusercontent.com/50994221/118699795-f4591180-b7df-11eb-9917-4bd4a455a89f.png">

This is our implemented radio button that we replaced <Datepicker> with in ../frontend/src/Cart.js

              <div class="control">
                <label class="radio">
                  <input type="radio" name="days" value="saturday"/>
                    Saturday (May 22)
                </label>
                <label class="radio">
                  <input type="radio" name="days" value="sunday"/>
                    Sunday (May 23)
                </label>
              </div>

This is our test function that outputs the user input (in this case we selected Sunday)

```
  function shoot() { //print function for testing
    var radios = document.getElementsByName('days');

    for (var i = 0, length = radios.length; i < length; i++) {
      if (radios[i].checked) {
        // do whatever you want with the checked radio
        setDeliveryTime(radios[i].value);
        alert(radios[i].value); // alert for testing
    
        // only one radio can be logically checked, don't check the rest
        break;
      }
    }
  }
```

<img width="703" alt="Screen Shot 2021-05-18 at 1 45 32 PM" src="https://user-images.githubusercontent.com/50994221/118713621-25d9d900-b7f0-11eb-8181-9e5de96c924f.png">

However, our MongoDB instead took in "saturday" which is our default value for setDeliveryTime. 

` const [deliveryTime, setDeliveryTime] = useState("saturday");
`

Any idea on how we can implement setDeliveryTime in our order button ("pay-btn") to update the MongoDB entry with the correct user input?

**Comment by AkashMullick** [comment]
We narrowed down our issue to `setDeliveryTime(radios[i].value);` not working in the way expect due to `useState` being asynchronous (as described in [this post](https://stackoverflow.com/questions/54069253/usestate-set-method-not-reflecting-change-immediately)). We tried following the suggested answers but we were not able to get the delivery time to update. Do you have any suggestions or ideas?
